src/stepfunctions/inputs/placeholders.py

```diff
@@ -291,3 +291,26 @@ def _create_variable(self, name, parent, type=None):
             return StepInput(name=name, parent=parent, type=type)
         else:
             return StepInput(name=name, parent=parent)
+
+
+class StepResult(Placeholder):
+
+    """
+        Top-level class for step result placeholders.
+    """
+
+    def __init__(self, schema=None, **kwargs):
+        super(StepResult, self).__init__(schema, **kwargs)
+        self.json_str_template = '${}'
+
+    def _create_variable(self, name, parent, type=None):
+        """
+            Creates a placeholder variable for Step Result.
+            A placeholder variable can only be created if the collection is not immutable due to a pre-specified schema.
+        """
+        if self.immutable:
+            raise ValueError("Placeholder variable does not conform to schema set for the placeholder collection.")
```

Comment:
nit: think of error messages as "help messages". this currently says what's broken, but would be most helpful if it guides users towards how they can fix it.

Reply:
I added the schema definition in the error message - this will be helpful to debug

It would look something like this:
```
ValueError: Placeholder variable does not conform to schema set for the placeholder collection: {'Name': <class 'str'>}
```
Changes included in the last commit